### DISCUSSION TITLE
In 8.0.7, Image.ToPlatform(Image.Handler.MauiContext) now gives 'Microsoft.Maui.Platform.WrapperView' rather than the actual platform types (UIImageView/ImageView/Image)

### DISCUSSION THREAD

**Discussion by jonmdev** [dotnet/maui#21669] [Q&A] [answered]
### Description

I just upgraded Visual Studio to access 8.0.7 and I notice now that the code:

```
Image imageView = new();
...
imageView.HandlerChanged += delegate {
    Debug.WriteLine("NEW IMAGE VIEW PLAT TYPE " + imageView.ToPlatform(imageView.Handler.MauiContext).GetType());
}
```

Now returns `Microsoft.Maui.Platform.WrapperView`. Previously I believe it gave the following actual native image view controls, which I was personally referencing as NativeImageViewType:

```
#if IOS
    global using NativeImageViewType = UIKit.UIImageView;
#elif ANDROID
    global using NativeImageViewType = Android.Widget.ImageView;
#elif WINDOWS
    global using NativeImageViewType = Microsoft.UI.Xaml.Controls.Image;
```

I note a workaround which still allows you to access these:

```
ImageHandler imageHandler = imageView.Handler as ImageHandler; 
NativeImageViewType nativeImageView = imageHandler.PlatformView;
```

But this seems strange. 

What is WrapperView and why does Image.ToPlatform return it? Are we supposed to retrieve the native image controls from inside that?

Thanks for any correction or fix.

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.3 GA

### Affected platforms

iOS, Android, Windows

**Comment by PureWeen** [comment, marked as answer]
WrapperView will sometimes get added when we have to add various clipping and interaction features. We have `ToPlatform` return the wrapper view because that's the view we are interested in when adding the view to the layouts. 

Generally 

```C#
ImageHandler imageHandler = imageView.Handler as ImageHandler; 
NativeImageViewType nativeImageView = imageHandler.PlatformView
```

Is preferred if you are just wanting to access the platformView. 

We mainly use "ToPlatform" when we are going to generate a platform view from an xplat view.
